test/compress/issue-203.js

```diff
@@ -26,9 +26,7 @@ compress_new_function_with_destruct: {
     }
     expect: {
         Function("a", "[b]", "return a");
-        Function("a", "{bb}", "return a");
-        Function("[[a]]", "[{bb}]", 'return a');
+        Function("a", "{bb:b}", "return a");
+        Function("[[a]]", "[{bb:b}]", 'return a');
```

Comment:
I'm surprised that uglify handles `Function` syntax at all.

That aside, why does it produce `b` at all without `mangle` enabled and why can't I reproduce this from the CLI with this PR?
```js
$ echo 'new Function("aa, {bb}", "return aa;");' | bin/uglifyjs -c unsafe,ecma=6 -b ecma=6
Function("aa, {bb}", "return aa;");
```